File: ValidPalindrome.py
class Solution(object):
    def isPalindrome(self, s):
        """
        :type s: str
        :rtype: bool
        """
        l = len(s)
        i,j = 0,l-1
        while i <j:
            while not s[i].isalnum() and i<j: 
                i += 1
            while not s[j].isalnum() and j>i:
                j -= 1
            if i>=j: break
            else:
                if s[i].lower() != s[j].lower():
                    return False
            i += 1
            j -= 1
        return True

File: validPalindrome.kt
class Solution {
    fun isPalindrome(s: String): Boolean {
        var l = 0
        var r = s.length -1
        do {
            while (l < r && !(s[l].isLetter() || s[l].isDigit() )) l++
            while (l < r && !(s[r].isLetter() || s[r].isDigit() )) r--
            if (l >= r) return true
            if (s[l].uppercaseChar() != s[r].uppercaseChar()) return false
            l++
            r--
        } while (l < r)
        return true
    }
}
